femtools/Read_GMSH.F90

```diff
@@ -290,14 +339,15 @@ end function read_gmsh_simple
   ! Read through the head to decide whether binary or ASCII, and decide
   ! whether this looks like a GMSH mesh file or not.
 
-  subroutine read_header( fd, lfilename, gmshFormat )
+  subroutine read_header( fd, lfilename, gmshFormat, versionNumber )
     integer fd, gmshFormat
 
     character(len=*) :: lfilename
     character(len=longStringLen) :: charBuf
     character :: newlineChar
-    integer gmshFileType, gmshDataSize, one
-    real versionNumber
+    integer :: gmshFileType, gmshDataSize, one, i
+    logical :: decimalVersion = .false.
```

Comment:
Again, something not adhered to in the rest of the module. But I think as a minimum code style guide for subroutines we try to use:
- there should be at least a single comment line on top telling what the subroutine does (unless completely obvious)
- all arguments should have intent declared - except pointer argument for which declaring intent has only been recently allowed in the fortran standard (and refers to whether the association status is changed, not changes into the thing that it points to)
- all arguments should be declared on top, before any other local variables

Reply:
I'll claim some innocence by saying I didn't write most of the code! These are, however, good conventions and I'll make sure this PR is following them.